proj/src/main/webapp/script.js

```diff
@@ -3,53 +3,69 @@
  */
 
 // Elements of login container
+const fname = document.getElementById("fname")
 const txtEmail = document.getElementById("email");
 const txtPassword = document.getElementById("pass");
 const btnLogin = document.getElementById("btnLogin");
 const btnSignUp = document.getElementById("btnSignUp");
 const btnLogout = document.getElementById("btnLogout");
 
 // Add login event
-btnLogin.addEventListener("click", e => {
-    const emailVal = txtEmail.value;
-    const passVal = pass.value;
+if(btnLogin){
+    btnLogin.addEventListener("click", e => {
+        const emailVal = txtEmail.value;
+        const passVal = pass.value;
 
-    // Initialize auth object
-    const auth = firebase.auth();
+        // Initialize auth object
+        const auth = firebase.auth();
 
-    const promise = auth.signInWithEmailAndPassword(emailVal, passVal);
-    promise.catch(e => console.log(e.message));
-    window.location.replace("chat.html");
-});
-
-// Add sign up event 
-btnSignUp.addEventListener("click", e => {
-    const emailVal = txtEmail.value;
-    const passVal = pass.value;
-
-    // Initialize auth object
-    const auth = firebase.auth();
-    auth.useDeviceLanguage();
+        const promise = auth.signInWithEmailAndPassword(emailVal, passVal).then(function(user){
+            window.location.replace("chat.html");
+        });
+        promise.catch(e => console.log(e.message));
+    });
+}
 
-    const promise = auth.createUserWithEmailAndPassword(emailVal, passVal);
-    promise.catch(e => console.log(e.message));
-    window.location.replace("chat.html");
+// Add sign up event
+if(btnSignUp){
+    btnSignUp.addEventListener("click", e => {
+        const emailVal = txtEmail.value;
+        const passVal = pass.value;
+
+        // Initialize auth object
+        const auth = firebase.auth();
+        auth.useDeviceLanguage();
+
+        const promise = auth.createUserWithEmailAndPassword(emailVal, passVal).then(function(){
+            auth.currentUser.updateProfile({displayName: fname.value + " " + lname.value}).then(function(){
+                console.log("display name updated successfully");
+                window.location.replace("chat.html");
+            }).catch(function(){
+                console.log("error updating display name");
+            })
+        });
+        promise.catch(e => console.log(e.message));
 
-});
+    });
+}
 
-btnLogout.addEventListener("click", e => {
-    firebase.auth().signOut();
-    window.location.replace("welcome.html");
-});
+if(btnLogout){
+    btnLogout.addEventListener("click", e => {
+        firebase.auth().signOut();
+        window.location.replace("welcome.html");
+    });
+}
 
 firebase.auth().onAuthStateChanged(firebaseUser => {
     if(firebaseUser){
         console.log(firebaseUser);
-        btnLogout.classList.remove("hidden");
+        if(btnLogout)
```

Comment:
out of curiosity what does adding this if statement do? is it checking if it's hidden or not?

Reply:
The old code removes a class "hidden" to the element, which switches its visibility state
The new code checks to see if it exists